### PR TITLE
Fix floating windows being hidden on OSX

### LINKED ISSUE #16790: Auto-Floating Windows on Mac
*This issue is based on feedback I received on the Mantid training course held at the MLZ in Munich June 30 - July 1.*

Mac users at the MLZ indicated that they kept losing the graph windows behind the main application window when plotting and performing fits. The only ways to solve this issue was to change the window to a docked window as hidden windows are not easily accessible. This does not occur on windows.

### PR DESCRIPTION
**Pull request by samueljackson92**
**Needs to be tested by a Mac user**

This fix prevents floating windows for disappearing behind the main Mantid application window.

**To test:**
- Go to Mantid > Preferences > General > Floating Windows and set everything to floating
- Create lots of different types of windows
- Check that they remain in front of the main window at all times.
- Check that they remain below any other open applications when Mantid is not focused.

Fixes #16790 .

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AntonPiccardoSelg** [comment]
Code looks fine.
Functional test:
* Followed the instructions with different window types
* Works as expected.

Happy to :shipit:

File: MantidPlot/src/FloatingWindow.cpp [hunk 1]
```diff
@@ -39,6 +39,14 @@ FloatingWindow::FloatingWindow(ApplicationWindow *appWindow, Qt::WindowFlags f)
   // Instead, the ApplicationWindow->removeFloatingWindow() call takes care of
   // calling deleteLater().
   setAttribute(Qt::WA_DeleteOnClose, false);
+
+#ifdef Q_OS_MAC
+  // Work around to ensure that floating windows remain on top of the main
+  // application window, but below other applications on Mac
+  Qt::WindowFlags flags = windowFlags();
+  Qt::WindowFlags new_flags = flags | Qt::Tool;
```
Comment (FedeMPouzols) [inline MantidPlot/src/FloatingWindow.cpp]: A few lines up (line 35) there is similar code for windows. They save the flags in `m_flags` which is then used in `FloatingWindow::event()` when there is a `WindowStateChange`. Would that be needed on osx as well?

**Comment by peterfpeterson** [comment]
@samueljackson92 should respond to @FedeMPouzols comment before this is merged.

File: MantidPlot/src/FloatingWindow.cpp (hunk 1 above)
Comment (samueljackson92) [inline MantidPlot/src/FloatingWindow.cpp]: I think you are correct. I shall fix that today.

Comment (FedeMPouzols) [inline MantidPlot/src/FloatingWindow.cpp]: I see. This must be a windows specific issue that when they maximize/minimize the window the flags are "lost" and need to be re-set manually.

**Pull request merged**